library/wait_for_mco_completion.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python
+
+from ansible.module_utils.basic import AnsibleModule
+import subprocess
+import time
+
+
+def run_command(command):
+    """Run a shell command and return its output or raise an error."""
+    try:
+        result = subprocess.run(
+            command, shell=True, text=True, capture_output=True, check=True
+        )
+        return result.stdout.strip(), None
+    except subprocess.CalledProcessError as err:
+        return None, Exception(f"Command '{command}' failed: {err.stderr.strip()}")
+
+
+def wait_for_mco(module, timeout):
+    """Wait until MCO conditions are satisfied or timeout."""
+    start_time = time.time()
+    interval = 3
+
+    while time.time() - start_time < timeout:
+        output, error = run_command("oc wait mcp --all --for=condition=UPDATED=True --timeout=60s && \
+                                     oc wait mcp --all --for=condition=UPDATING=False --timeout=60s && \
+                                     oc wait mcp --all --for=condition=DEGRADED=False --timeout=60s")
```

Comment:
we have had to tweak the timeout in the past for MTU migration, better to make it a parameter.

Reply:
okay, well maybe 60 is okay.  for MTU we switched from 10s to 30s.  https://github.com/openshift/release/pull/44285